### PR TITLE
WebUI tests: Extend netgroup tests with more scenarios

### PR DESCRIPTION
**Pull request by varunmylaraiah**
Extended webui group automation test with below scenarios
    Scenarios
     *add netgroup with invalid names
     *add and delete records in various scenarios
     *verify button's action in various scenarios.
    
    https://pagure.io/freeipa/issue/7505

### CODE REVIEW

**Comment by Rezney** [comment]
Travis does not like these (no newline at end of file already commented):

`./ipatests/test_webui/data_netgroup.py:100:80: E501 line too long (80 > 79 characters)
./ipatests/test_webui/data_netgroup.py:102:2: W292 no newline at end of file
./ipatests/test_webui/test_netgroup.py:197:51: E128 continuation line under-indented for visual indent`

**Comment by Rezney** [comment]
@pvoborni could you please also take a look? Thanks...

**Comment by varunmylaraiah** [comment]
Noticed PEP-8 errors. Will fix it

File: ipatests/test_webui/test_netgroup.py [hunk 1]
```diff
@@ -42,6 +47,160 @@ def test_crud(self):
...
+        self.assert_record(pkey)
+
+        # Negative search
+        pkey = netgroup.DATA5['pkey']
+        self.search_pkey(pkey)
+        self.assert_record(pkey, negative=True)
+
+        # delete multiple records
+        records = [netgroup.DATA, netgroup.DATA2, netgroup.DATA3]
+        self.select_multiple_records(netgroup.ENTITY, records)
```
[the remark this answers is not in the document]
Comment (pvoborni) [reply ipatests/test_webui/test_netgroup.py]: Seems that the only Web UI test failure is being caused by this.

[the remark this answers is not in the document]
Comment (varunmylaraiah) [reply ipatests/test_webui/test_netgroup.py]: Fixed

**Comment by pvoborni** [comment]
In PR #1838 I made a comment:

>  I'm beginning to think whether we should stop using general DATAX names for data which serve for a specific test. Originally there was only a need for more records without a special meaning to test the simplest CRUD or add more records as members.

> Here if somebody changes the data in future, we could easily miss its purpose - a test case. So e.g. instead of DATA, it can be DATA_SPECIAL_CHARS. Same with other DATAs.

I think it applies here as well. DATA5, DATA6 and DATA7 have a special meaning. So they should be renamed to avoid future problem e.g. to DATA_MIXED_CASE, DATA_LONG_NAME, DATA_SINGLE_CHAR

Otherwise, this PR looks more or less good to me. I discussed with Michal whether `test_add_netgroup` describes what it does the best. Maybe `test_basic_workflows` would be a better description. 

Other my comments would just nitpicks about various naming and styling, but they are not that necessary and are more a personal preferences so we don't need to block this PR with those.

In summary:
- [ ] rename DATA5, DATA6 and DATA7 
- [ ] rename test_add_netgroup
 
And then it's green from me.

**Comment by pvoborni** [comment]
Travis will fail on pylint:
```
pycodestyle
-----------
./ipatests/test_webui/test_netgroup.py:67:80: E501 line too long (80 > 79 characters)
make: *** [Makefile:1174: fastlint] Error 1

```

**Comment by Rezney** [comment]
master:

* 00a8d00ea9f3ac2ce595670f2791b9b6bf20f5e4 WebUI tests: Extend netgroup tests with more scenarios
